MovieApp/MoviePictureCell.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  MovieCatergoyCell.swift
+//  MovieApp
+//
+//  Created by endava-bootcamp on 02.05.2023..
+//
+
+import Foundation
+import UIKit
+import PureLayout
+import MovieAppData
+
+class MoviePictureCell: UICollectionViewCell {
+    static let identifier = "MoviePictureCell"
+    
+    var navigationController: UINavigationController!    
+    var img : UIImage!
+    var imgView: UIImageView!
+    var likeButton : UIButton!
+    var id : Int!
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        buildViews()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func buildViews() {
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+        self.autoSetDimensions(to: CGSize(width: self.frame.width, height: self.frame.height))
+        print("cell")
+        print(self.frame.height)
+
+    }
+
+    private func createViews() {
+        img = UIImage()
+        imgView = UIImageView()
+        contentView.addSubview(imgView)
+        
+        likeButton = UIButton()
+        contentView.addSubview(likeButton)
+    }
+    
+    private func styleViews() {
+        imgView.image = img
+        imgView.clipsToBounds = true
+        imgView.contentMode = .scaleAspectFill
+        
+        likeButton.frame = CGRect(x: 0, y: 0, width: 32, height: 32)
+        likeButton.backgroundColor = UIColor(hex: 0x0B253F).withAlphaComponent(0.6)
+        likeButton.layer.cornerRadius = 16
+        likeButton.setImage(UIImage(systemName: "heart")?.withTintColor(.white, renderingMode: .alwaysOriginal), for: .normal)
+        likeButton.contentEdgeInsets = UIEdgeInsets(top: 5, left: 5, bottom: 5, right: 5)
+        likeButton.layer.cornerRadius = likeButton.bounds.size.height/2
+        likeButton.autoSetDimensions(to: likeButton.frame.size)
+        likeButton.clipsToBounds = true
```

Comment:
As mentioned before, try not to set the frames directly but use PureLayout to do so

in autoSetDimensions you could've just set the dimension. Also there is a lot of duplicated code here

```swift
        likeButton.backgroundColor = UIColor(hex: 0x0B253F).withAlphaComponent(0.6)
        likeButton.setImage(UIImage(systemName: "heart")?.withTintColor(.white, renderingMode: .alwaysOriginal), for: .normal)
        likeButton.contentEdgeInsets = UIEdgeInsets(top: 5, left: 5, bottom: 5, right: 5)
        likeButton.layer.cornerRadius = 32/2
        likeButton.clipsToBounds = true
```

and put `likeButton.autoSetDimensions(to: likeButton.frame.size)` in define layout for views since it defines the constraint for the width and height